src/nifgen/examples/nifgen_script.py

```diff
@@ -0,0 +1,185 @@
+
+
+def calculate_rampup():
+    ramp = np.arange(0, 0.5, 0.01)                # numpy.arange([start,]stop, [step,]dtype=None(
+    ramp_up = ramp.tolist()                       # List of Float
+    return ramp_up
+
+
+def calculate_rampdown():
+    ramp = np.arange(0, 0.5, 0.01)                # numpy.arange([start,]stop, [step,]dtype=None)
+    ramp_down = ramp.tolist()                     # List of Float
+    ramp_down.reverse()                           # Reverse list to get a ramp down
+    return ramp_down
+
+
+def calculate_square():
+    time = np.arange(0, 10, 0.01)                 # np.arange(Start, Stop, Step)
+    square_build = signal.square(time, 0.5)       # signal.square(time,duty_cycle)
+    square = square_build.tolist()                # List of Float
+    return square
+
+
+def calculate_triangle():
+    time = np.linspace(0, 1, 200)                 # np.arange(Start, Stop, Step)
+    triangle_build = signal.sawtooth(time)        # signal.square(time,width)
+    triangle = triangle_build.tolist()            # List of Float
+    return triangle
+
+
+def calculate_gaussian_noise():
+    random_noise = np.random.normal(0, 0.1, 300)  # random.normal(center, scale, size)
+    noise = random_noise.tolist()
+    return noise
+
+
+script_all = '''
+script scriptmulti
+  repeat until scriptTrigger0
+    Generate rampup
+    Generate sine
+    Generate rampdown
+  end repeat
+  repeat until scriptTrigger0
+    Generate rampdown
+    Generate square
+    Generate rampup
+  end repeat
+  repeat until scriptTrigger0
+    Generate rampup
+    Generate rampdown
+  end repeat
+  repeat until scriptTrigger0
+    Generate sine
+  end repeat
+  repeat until scriptTrigger0
+    Generate triangle
+  end repeat
+  repeat until scriptTrigger0
+    Generate rampdown
+    Generate noise
+    Generate rampup
+  end repeat
+end script
+
+script scriptsine
+  repeat until scriptTrigger0
+    Generate sine
+  end repeat
+end script
+
+script scriptrampup
+  repeat until scriptTrigger0
+    Generate rampup
+  end repeat
+end script
+
+script scriptrampdown
+  repeat until scriptTrigger0
+    Generate rampdown
+  end repeat
+end script
+
+script scriptsquare
+  repeat until scriptTrigger0
+    Generate square
+  end repeat
+end script
+
+script scripttriangle
+  repeat until scriptTrigger0
+    Generate triangle
+  end repeat
+end script
+
+script scriptnoise
+  repeat until scriptTrigger0
+    Generate noise
+  end repeat
+end script
+'''
+
+
+def example(resource_name, options, shape, channel):
+    with nifgen.Session(resource_name=resource_name, options=options, channel_name=channel) as session:
+        # CONFIGURATION
+        # 1 - Set the mode to Script
+        session.output_mode = nifgen.OutputMode.SCRIPT
+
+        # 2 - Configure Trigger:
+        # SOFTWARE TRIGGER: used in the script
+        session.script_triggers[0].script_trigger_type = nifgen.ScriptTriggerType.SOFTWARE_EDGE  # TRIG_NONE / DIGITAL_EDGE / DIGITAL_LEVEL / SOFTWARE_EDGE
```

Comment:
API Question: What does it mean to `[0]` on `script_triggers`?

Are there a predefined set of script triggers and this is just changing the values away from default?

Reply:
There are 4 script triggers, 0 through 3.
In the script they are referred to as ScriptTrigger0 and so forth.

Reply:
Yes, see the scripts defined in `SCRIPTS_ALL`